File: Data Structures & Algorithms/jump-game/submission-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        vector<bool>dp(nums.size(),false);
        int n=nums.size();
        dp[nums.size()-1]=true;
        for(int i=n-2;i>=0;i--){
            bool f=false;
            for(int j=1;j<=nums[i] && j<n;j++){
            if(dp[i+j]) {
                dp[i]=true;
                f=true;
            }
        }
        if(!f)dp[i]=false;
        }
        return dp[0];
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-1.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n=len(nums)
        dp=[float('inf')]*(n)
        dp[n-1]=0
        for i in range(n-2,-1,-1):
            end=min(n-1,i+nums[i])
            res=float('inf')
            for j in range(i+1,end+1):
                res=min(res,1+dp[j])
            dp[i]=res

        return dp[0]

